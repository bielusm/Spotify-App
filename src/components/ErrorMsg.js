import React from "react";
import { connect } from "react-redux";

export const ErrorMsg = props => (
  <>
    {props.errMsg && (
      <h2 id="errMsg" className="my-4">
        {props.errMsg}
      </h2>
    )}
  </>
);

const mapStateToProps = state => ({ errMsg: state.error });

export default connect(mapStateToProps)(ErrorMsg);
