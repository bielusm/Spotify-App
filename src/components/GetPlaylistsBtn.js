import React from "react";

const GetPlaylistsBtn = props => (
  <button
    id="getPlaylistsBtn"
    className="button"
    onClick={props.getPlaylists}
    disabled={props.getPlaylistsDisabled}
  >
    Pick playlists
  </button>
);

export default GetPlaylistsBtn;
