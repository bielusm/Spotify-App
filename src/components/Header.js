import React from "react";

const Header = () => <h1>Spotify Manager</h1>;
export default Header;
