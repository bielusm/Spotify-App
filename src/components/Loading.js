import React from "react";

export const Loading = props => (
  <div>
    {props.loading && (
      <div className="loadingStatus">
        <div className="spinner-border" role="status">
          <span className="sr-only"></span>
        </div>
        <h2 className="mx-2 d-inline">Loading...</h2>
      </div>
    )}
  </div>
);

export default Loading;
