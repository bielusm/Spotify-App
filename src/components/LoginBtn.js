import React from "react";

export default class LoginBtn extends React.Component {
  login = () => {
    const scope =
      "playlist-modify-public%20user-read-playback-state%20playlist-read-private%20playlist-modify-private";
    const redirect_uri = window.location.origin;
    //redirect_uri = redirect_uri.slice(0, redirect_uri.length - 1);
    const url = `https://accounts.spotify.com/authorize?client_id=4252feb807d04ced962e15f346258957&response_type=token&redirect_uri=${redirect_uri}&scope=${scope}&show_dialog=false`;
    window.location.href = url;
  };

  render() {
    return (
      <div>
        {this.props.loginVisible && (
          <button id="login" className="btn btn-secondary" onClick={this.login}>
            Log in
          </button>
        )}
      </div>
    );
  }
}
