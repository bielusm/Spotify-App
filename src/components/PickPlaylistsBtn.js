import React from "react";
import Button from "@material-ui/core/Button";

export const PickPlaylistsBtn = props => (
  <Button
    variant="contained"
    color="primary"
    id="pickPlaylistsBtn"
    onClick={props.getPlaylists}
    disabled={props.getPlaylistsDisabled}
  >
    Pick playlists
  </Button>
);

export default PickPlaylistsBtn;
