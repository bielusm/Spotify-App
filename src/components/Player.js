import React from "react";
import { Button, IconButton } from "@material-ui/core";

import {
  modifyLibrary,
  startPlayer,
  pausePlayer,
  nextTrack,
  prevTrack
} from "../api/player";

import { setErrorMsg } from "../store/actions/error";
import { connect } from "react-redux";

export class Player extends React.Component {
  togglePause = async () => {
    if (this.props.is_playing)
      pausePlayer(this.props.bearerToken).then(() => this.props.update());
    else startPlayer(this.props.bearerToken).then(() => this.props.update());
  };

  nextTrack = async () => {
    nextTrack(this.props.bearerToken).then(() => this.props.update());
  };

  prevTrack = async () => {
    prevTrack(this.props.bearerToken).then(() => {
      this.props.update();
    });
  };

  toggleLike = async () => {
    modifyLibrary(
      this.props.bearerToken,
      this.props.id,
      this.props.likes_song
    ).then(() => this.props.update());
  };
  render() {
    return (
      <>
        <IconButton
          className="prevBtn fas fa-backward"
          onClick={this.prevTrack}
        />

        <IconButton
          className={
            "playBtn fas " + (this.props.is_playing ? "fa-pause" : "fa-play")
          }
          onClick={this.togglePause}
        />
        <IconButton
          className="nextBtn fas fa-forward"
          onClick={this.nextTrack}
        />
        <IconButton
          className={
            "likeBtn " +
            (this.props.likes_song ? "fas fa-heart" : "far fa-heart")
          }
          onClick={this.toggleLike}
        />
      </>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setErrorMsg: error => dispatch(setErrorMsg(error))
});

export default connect(undefined, mapDispatchToProps)(Player);
