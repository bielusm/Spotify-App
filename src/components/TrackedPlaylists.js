import React from "react";

const TrackedPlaylists = props => (
  <div>
    {props.trackedPlaylists.length > 0 && (
      <div id="trackedPlaylists">
        <h2>Tracked Playlists</h2>
        <ul className="list-group">
          {props.trackedPlaylists.map(playlist => {
            const plName = playlist.name.replace(/ /g, "-");
            return (
              <li
                key={playlist.id}
                id={playlist.id}
                className={
                  "d-flex justify-content-between list-group-item " + plName
                }
              >
                {playlist.name}
                {playlist.found ? (
                  <button
                    className="btn btn-danger remove"
                    onClick={props.addOrRemove}
                  >
                    Remove
                  </button>
                ) : (
                  <button
                    className="btn btn-success add"
                    onClick={props.addOrRemove}
                  >
                    Add
                  </button>
                )}
              </li>
            );
          })}
        </ul>
      </div>
    )}
  </div>
);

export default TrackedPlaylists;
