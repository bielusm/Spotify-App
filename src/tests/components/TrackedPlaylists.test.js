import React from "react";
import { shallow } from "enzyme";
import { TrackedPlaylists } from "../../components/TrackedPlaylists";
import {
  trackedPlaylists as trackedPlaylistsFixture,
  currentTrack as currentTrackFixture
} from "../fixtures/trackedPlaylists";
import { Button } from "@material-ui/core";

let wrapper, trackedPlaylists, currentTrack, addOrRemove;

beforeEach(() => {
  trackedPlaylists = trackedPlaylistsFixture;
  currentTrack = currentTrackFixture;
  addOrRemove = jest.fn();
  wrapper = shallow(
    <TrackedPlaylists
      currentTrack={currentTrack}
      trackedPlaylists={trackedPlaylists}
      addOrRemove={addOrRemove}
    />
  );
});

test("Tracked playlist snapshot test", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Tracked playlist snapshot test no data", () => {
  wrapper.setProps({ trackedPlaylists: [] });
  expect(wrapper).toMatchSnapshot();
});

test("Test add or remove click", () => {
  wrapper
    .find(Button)
    .at(0)
    .simulate("click");
  expect(addOrRemove).toHaveBeenCalled();
});
