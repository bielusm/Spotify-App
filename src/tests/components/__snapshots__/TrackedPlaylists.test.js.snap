// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracked playlist snapshot test 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Table))
    id="trackedPlaylists"
  >
    <WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Typography))
            className="trackedPlaylists-header"
          >
            Tracked Playlists
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow))
        className="trackedPlaylists-list_item Big-rap-playlist"
        id="4TL6CsXn7AFFrgMWMxq1XL"
        key="4TL6CsXn7AFFrgMWMxq1XL"
      >
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Typography))
            className="trackedPlaylists-list_item_text"
            variant="p"
          >
            Big rap playlist
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Button))
            className="button button-add add trackedPlaylists-list-button"
            color="primary"
            onClick={[MockFunction]}
            variant="contained"
          >
            Add
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        className="trackedPlaylists-list_item everything"
        id="6R4CXv7L5mETXDD3z9360j"
        key="6R4CXv7L5mETXDD3z9360j"
      >
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Typography))
            className="trackedPlaylists-list_item_text"
            variant="p"
          >
            everything
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Button))
            className="button button-add add trackedPlaylists-list-button"
            color="primary"
            onClick={[MockFunction]}
            variant="contained"
          >
            Add
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        className="trackedPlaylists-list_item :O"
        id="2lcW5jiLBwniHL8CyKg1H8"
        key="2lcW5jiLBwniHL8CyKg1H8"
      >
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Typography))
            className="trackedPlaylists-list_item_text"
            variant="p"
          >
            :O
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(Button))
            className="button button-add add trackedPlaylists-list-button"
            color="primary"
            onClick={[MockFunction]}
            variant="contained"
          >
            Add
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
</Fragment>
`;

exports[`Tracked playlist snapshot test no data 1`] = `<Fragment />`;
